src/ir.h

```diff
@@ -135,6 +141,9 @@ struct Expr {
   static Expr* CreateBrTable(VarVector* targets, Var default_target);
   static Expr* CreateCall(Var);
   static Expr* CreateCallIndirect(Var);
+  static Expr* CreateCatch(Var);
+  static Expr* CreateCatchAll();
+  static Expr* CreateCatchBlock(Expr* catch_, Expr* first);
```

Comment:
First, they are not a block (in the same way that an "else" of the "if" isn't a block).

Changing to have catch and catch_all be a { var , expr* } pair.